fuzz/CMakeLists.txt

```diff
@@ -16,10 +16,7 @@ macro(fuzzer name)
                     COMPILE_FLAGS "-fsanitize=fuzzer"
                     LINK_FLAGS "-fsanitize=fuzzer")
     endif()
-    target_link_libraries(${name}
-            PRIVATE
-            exiv2lib
-            )
+    target_link_libraries(${name} PRIVATE exiv2lib std::filesystem)
```

Comment:
Instead of explicit std::filesystem here, isn't this a matter of linking the correct C++ library - libc++ for clang should have these symbols AFAIK, perhaps the linker is trying to use the default (and older) GNU's libstdc++ instead?

I.e. perhaps we need to check if we have a proper CMake toolchain file for Clang on Linux defining stuff like https://gitlab.kitware.com/cmake/cmake/-/issues/18275

Reply:
Actually I see we already have detection for libc++ here https://github.com/Exiv2/exiv2/blob/15c91b4cc9b50197c83a94ab3150d4709fd09bb5/cmake/FindFilesystem.cmake#L227 and after reading https://gitlab.kitware.com/cmake/cmake/-/issues/17834 it seems explicit linking is still required, so ignore this...

Reply:
Very good point.

When I have used clang in other computers I was using the clang binaries provided by the OS packaging system and probably they were making use of libstdc++ by default (On recent versions of Ubuntu). In this case I was using a pre-compiled version which I just installed into my `/opt` folder and I did not consider the common issues regarding the C++ library. 

It is relevant to note that the CMake guys are thinking about introducing a new family of functions to be able to configure the standard library to use:
https://gitlab.kitware.com/cmake/cmake/-/issues/20851

Since this target is only intended to be compiled when using clang, I think that the change does not make sense anymore. If I ever work on a CMake toolchain file for this configuration that could be useful for the community I will think about including it in the repository.

Thanks for your feedback!

Reply:
I just read your 2nd comment. Ok, since it is anyway "fixing/helping" to a particular configuraiton without affecting others (so far), we can try to leave it.

I'll drop anyways the 2nd commit about replacing the deprecated calls to gtest.